File: spec/e2e/form_post_spec.rb
# frozen_string_literal: true

require_relative "e2e_helper"

RSpec.describe "response_mode=form_post" do
  it "delivers the code via POST and the exchange still succeeds" do
    client = dashboard_client("formpost")
    run_auth_code_flow(client, response_mode: "form_post", nonce: true)

    expect(page).to have_css("#exchange_result", text: '"id_token"')
  end
end

File: e2e/tests/form-post.spec.ts
import { test, expect } from '@playwright/test';
import { setup, runAuthCodeFlow } from './helpers';

test.describe('response_mode=form_post', () => {
  test('delivers the code via POST and the exchange still succeeds', async ({ page }) => {
    const s = await setup(page, 'formpost');
    await runAuthCodeFlow(page, s, { responseMode: 'form_post', nonce: true });
    await expect(page.locator('#exchange_result')).toContainText('"id_token"');
  });
});
